nemo/collections/asr/data/data_simulation.py

```diff
@@ -0,0 +1,1324 @@
+        if len(increase_percent) > 0:
+            # extract original per-speaker probabilities
+            dominance = np.copy(base_speaker_dominance)
+            for i in range(len(dominance) - 1, 0, -1):
+                dominance[i] = dominance[i] - dominance[i - 1]
+            # increase specified speakers by the desired factor
+            for i in increase_percent:
+                dominance[i] = dominance[i] * factor
+            # renormalize
+            dominance = dominance / np.sum(dominance)
+            for i in range(1, len(dominance)):
+                dominance[i] = dominance[i] + dominance[i - 1]
+            enforce = True
+        else:  # no unrepresented speakers, so enforce mode can be turned off
+            dominance = base_speaker_dominance
+            enforce = False
+        return dominance, enforce
+
+    def _set_speaker_volume(self):
+        """
+        Set the volume for each speaker
+        """
+        if self._params.data_simulator.session_params.normalization_type == 'equal':
+            self._volume = np.ones(self._params.data_simulator.session_config.num_speakers)
+        elif self._params.data_simulator.session_params.normalization_type == 'variable':
+            self._volume = np.random.normal(
+                loc=1.0,
+                scale=self._params.data_simulator.session_params.normalization_var,
+                size=self._params.data_simulator.session_config.num_speakers,
+            )
+            for i in range(0, self._params.data_simulator.session_config.num_speakers):
+                if self._volume[i] < self._params.data_simulator.session_params.min_volume:
+                    self._volume[i] = self._params.data_simulator.session_params.min_volume
+                if self._volume[i] > self._params.data_simulator.session_params.max_volume:
+                    self._volume[i] = self._params.data_simulator.session_params.max_volume
+
+    def _get_next_speaker(self, prev_speaker, dominance):
+        """
+        Get next speaker (accounting for turn probability, dominance distribution)
+
+        Args:
+            prev_speaker (int): Previous speaker turn.
+            dominance (list): Dominance values for each speaker.
+        Returns:
+            prev_speaker/speaker_turn (int): Speaker turn
+        """
+        if np.random.uniform(0, 1) > self._params.data_simulator.session_params.turn_prob and prev_speaker != None:
+            return prev_speaker
+        else:
+            speaker_turn = prev_speaker
+            while speaker_turn == prev_speaker:  # ensure another speaker goes next
+                rand = np.random.uniform(0, 1)
+                speaker_turn = 0
+                while rand > dominance[speaker_turn]:
+                    speaker_turn += 1
+            return speaker_turn
+
+    def _get_window(self, window_amount, start=False):
+        """
+        Get window
```

Comment:
Please add more description by saying something like : Construct window curve to alleviate abrupt change of time-series signal